src/taipy/core/data/_data_model.py

```diff
@@ -24,7 +24,7 @@ class _DataNodeModel:
     id: str
     config_id: str
     scope: Scope
-    storage_type: str
+    storage_type: Optional[str]
```

Comment:
**note to reviewers** The type is changed to `Optional[str]` to reflect the change of all `storage_type()` getters everywhere.

Reply:
I believe the storage_type cannot be optional but must be required. 
A data node without a storage type is inconsistent.

Reply:
Ok, let's change the implementation so that this method returns a mandatory `str` or raises an exception.